src/main/java/org/owasp/benchmark/score/parsers/SnappyTickReader.java

```diff
@@ -0,0 +1,149 @@
+ * @author Parthi Shah <a href="http://www.iappsecure.com">iAppSecure</a>
+ * @created 2016
+ *
+ * This file reuses existing OWASP Benchmark Project code with Fusion Lite Insight specific changes by Parthi Shah
+ */
+
+package org.owasp.benchmark.score.parsers;
+
+import java.util.List;
+
+import org.owasp.benchmark.score.BenchmarkScore;
+import org.w3c.dom.Node;
+
+public class SnappyTickReader extends Reader {
+
+	
+	public TestResults parse( Node root ) throws Exception {
+        
+		// root Node is Report
+		Node toolInfo = getNamedChild("ToolInfo", root);
+		Node tool = getNamedChild("Tool", toolInfo);
+        String toolName = getAttributeValue("Name", tool);
+        TestResults tr = new TestResults(toolName, true, TestResults.ToolType.SAST);
+
+        String version = getAttributeValue("Version", tool);
+        tr.setToolVersion(version);
+        
+		Node projectInfo = getNamedChild("ProjectInfo", root);
+		Node project = getNamedChild("Project", projectInfo);
+        String duration = getAttributeValue("Duration", project);
+        tr.setTime( duration );
+        
+        // Inside of the <VulnerabilityReport> are <Severity Level> then <VulnerabilityCollection> like so:
+        //<VulnerabilityReport>
+        //  <Severity Level="Critical">
+        //    <VulnerabilityCollection>
+        //      <Vulnerability Title="SQL Injection" CWE="89" OWASP="A1">
+        //        <Description>...</Description>
+        //        <Impact>...</Impact>
+        //        <Remediation>...</Remediation>
+        //        <FindingsList>
+        //          <Finding FileName="BenchmarkTest00008.java" LineNo="59" 
+        //                  CodeLine="java.sql.ResultSet rs = statement.executeQuery();" />      
+        
+        Node vulnReport = getNamedChild("VulnerabilityReport", root);
+        
+        // Loop through all the Severity nodes
+        List<Node> sevLevels = getNamedChildren("Severity", vulnReport);
+        
+        for (Node issue : sevLevels) {
+        	// There is only 1 VulnerabilityCollection per severity level
+            Node vulnCollect = getNamedChild("VulnerabilityCollection", issue);
+            
+            // There can be multiple Vulnerability nodes per VulnerabilityCollection
+            List<Node> vulnerabilities = getNamedChildren("Vulnerability", vulnCollect);
+            for (Node vulnerability : vulnerabilities) {
+	            String cweNum = getAttributeValue("CWE", vulnerability);
+	            int findingCWE = cweLookup(cweNum);
+	            // There is a single FindingsList per Vulnerability category
+	            Node findingsList = getNamedChild("FindingsList", vulnerability);
+	            List<Node> findings = getNamedChildren("Finding", findingsList);
+	            for (Node finding : findings) {
+	                String filename = getAttributeValue("FileName", finding);
+	                String findingName = filename.substring(0, filename.indexOf("."));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the indexOf() search on line 79 does not find the substring and returns -1, the call to substring() on line 79 will throw IndexOutOfBoundsException.